plugins/common/common.go

```diff
@@ -0,0 +1,149 @@
+package common
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	sh "github.com/codeskyblue/go-sh"
+)
+
+// DokkuCmd represents a shell command to be run for dokku
+type DokkuCmd struct {
+	Env           map[string]string
+	Command       *exec.Cmd
+	CommandString string
+	Args          []string
+	ShowOutput    bool
+}
+
+// NewDokkuCmd creates a new DokkuCmd
+func NewDokkuCmd(command string) *DokkuCmd {
+	items := strings.Split(command, " ")
+	cmd := items[0]
+	args := items[1:]
+	return &DokkuCmd{
+		Command:       exec.Command(cmd, args...),
+		CommandString: command,
+		Args:          args,
+		ShowOutput:    true,
+	}
+}
+
+// Execute is a lightweight wrapper around exec.Command
+func (dc *DokkuCmd) Execute() bool {
+	env := os.Environ()
+	for k, v := range dc.Env {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
+	dc.Command.Env = env
+	if dc.ShowOutput {
+		dc.Command.Stdout = os.Stdout
+		dc.Command.Stderr = os.Stderr
+	}
+	err := dc.Command.Run()
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// VerifyAppName verifies app name format and app existence"
+func VerifyAppName(appName string) bool {
+	dokkuRoot := GetEnv("DOKKU_ROOT")
+	appRoot := strings.Join([]string{dokkuRoot, appName}, "/")
+	_, err := os.Stat(appRoot)
+	if os.IsNotExist(err) {
+		LogFail(fmt.Sprintf("App %s does not exist", appName))
+	}
+	r, _ := regexp.Compile("^[a-z].*")
+	if !r.MatchString(appName) {
+		LogFail("App name must begin with lowercase alphanumeric character")
+	}
+	return true
+}
+
+// GetEnv returns env variable or fails if it's not set
+func GetEnv(key string) string {
+	dokkuRoot := os.Getenv(key)
+	if dokkuRoot == "" {
+		LogFail(fmt.Sprintf("%s not set!", key))
+	}
+	return dokkuRoot
+}
+
+// LogFail is the failure log formatter
+// prints text to stderr and exits with status 1
+func LogFail(text string) {
+	fmt.Fprintln(os.Stderr, fmt.Sprintf("FAILED: %s", text))
+	os.Exit(1)
+}
+
+// GetDeployingAppImageName returns deploying image identifier for a given app, tag tuple. validate if tag is presented
+func GetDeployingAppImageName(args ...string) (imageName string) {
```

Comment:
@mattaitchison @josegonzalez i'm curious what's wrong with the `GetDeployingAppImageName(args ...string)` implementation. can either of you expound a bit beyond "it's not idiomatic golang"? 😄